metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/aop/ReleaseDateAspect.java

```diff
@@ -72,13 +81,38 @@ public Object filterOnReleaseDateAdviceFindOne(ProceedingJoinPoint proceedingJoi
     }
 
     private Object getObjectUnlessBeforeReleaseDate(Object result) {
-        if (result instanceof Auditable) {
-            LocalDate releaseDate = ((Auditable) result).getReleaseDate();
-            if (releaseDate != null && releaseDate.isAfter(LocalDate.now())) {
-                return null;
-            }
+        if (!(result instanceof Auditable)) {
+            return result;
+        }
+
+        LocalDate releaseDate = ((Auditable) result).getReleaseDate();
+        if (releaseDate == null || !releaseDate.isAfter(LocalDate.now())) {
+            return result;
+        }
+
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if (authentication instanceof AnonymousAuthenticationToken) {
+            throw new SecurityException("Authentication required");
+        }
+
+        //Allow admins to access before release date
+        if (authentication.getAuthorities().stream()
+                .anyMatch(authority -> authority.getAuthority()
+                        .equals("ROLE_" + EnableSecurityConfig.ROLE_SERVICE_OPERATOR))) {
+            return result;
+        }
+
+        String commaSeperatedPermittedStudiesToTheUsers = ((CustomUsernamePasswordAuthenticationToken)
+                ((OAuth2Authentication) authentication).getUserAuthentication()).getStudies();
+        List<String> permittedStudies = (commaSeperatedPermittedStudiesToTheUsers.isEmpty()) ? new ArrayList<>() : Arrays.asList
+                (commaSeperatedPermittedStudiesToTheUsers.split(","));
```

Comment:
This string is long and bulky; would be better to rewrite it as a simple if-else statement (not the ternary if)